frontend/src/metabase-types/api/dashboard.ts

```diff
@@ -86,6 +86,7 @@ export type UnrestrictedLinkEntity = {
   display_name?: string;
   description?: string;
   display?: CardDisplayType;
+  getIcon?: () => { name: string };
```

Comment:
I believe the `getIcon` type shouldn't live under `metabase-types/api` as it doesn't come from the backend. Is there another place we can move it to? Should we consider adding `metabase/entities/types` or `metabase/lib/entities/types`? It can work like that:

```ts
import type { EntityWrapper } from "metabase/lib/entities"
import type { Dashboard } from "metabase-types/api"

type WrappedDashboard = Dashboard & EntityWrapper;

// Or maybe make every entity define the `Wrapped*` type:
// metabase/entities/dashboards.ts
import { createEntity, WrappedEntity } from "metabase/lib/entities"
import type { Dashboard } from "metabase-types/api"

const Dashboards = createEntity({ ... });

export type WrappedDashboard = Dashboard & WrappedEntity
export default Dashboards;
```

Reply:
Instead of making a fundamental change to how/where we store types, I'll create a little local union type for the wrapped entities, since they shouldn't be used outside this directory anyway.